DataDog/datadog-operator
# service discovery: Add operator support

### What does this PR do?

Adds service-discovery check to the operator.

### Motivation

Allow to control service-discovery from the operator.

### Additional Notes

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: v7.58.0
* Cluster Agent: N/A

### Describe your test plan

* Create `kind` cluster
* Follow instructions installations instructions https://github.com/DataDog/datadog-operator/blob/main/docs/how-to-contribute.md#testing-the-operator-for-development
* Use the following configuration
  ```yaml
  apiVersion: datadoghq.com/v2alpha1
  kind: DatadogAgent
  metadata:
    name: datadog
  spec:
    global:
      env:
        - name: DD_HOSTNAME
          value: "example-kind"
      clusterName: my-example-cluster
      credentials:
        apiSecret:
          secretName: datadog-secret
          keyName: api-key
        appSecret:
          secretName: datadog-secret
          keyName: app-key
    features:
      serviceDiscovery:
        enabled: true
    override:
      nodeAgent:
        image:
          tag: 7.59.0-linux-jmx
    ```
* Use `api key` to ddstaging
* Wait 5m and check you have entries [here](https://ddstaging.datadoghq.com/ddsql/editor?q=SELECT%20%0A%20%20%20%20a.%2A%0AFROM%20%0A%20%20%20%20discovered_service%20AS%20a%0Awhere%20a.host_name%20%3D%20%27example-kind%27%0AORDER%20BY%20%0A%20%20%20%20a.start_time&query_timeframe_end=1731222219529&query_timeframe_mode=sliding&query_timeframe_start=1730617419529)

### Checklist
